storage/api/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteStorageManager.java

```diff
@@ -107,20 +110,26 @@ InputStream fetchLogSegment(RemoteLogSegmentMetadata remoteLogSegmentMetadata,
      * @param endPosition              end position of log segment to be read, inclusive.
      * @return input stream of the requested log segment data.
      * @throws RemoteStorageException          if there are any errors while fetching the desired segment.
-     * @throws RemoteResourceNotFoundException when there are no resources associated with the given remoteLogSegmentMetadata.
+     * @throws RemoteResourceNotFoundException the requested log segment is not found in the remote storage.
      */
     InputStream fetchLogSegment(RemoteLogSegmentMetadata remoteLogSegmentMetadata,
                                 int startPosition,
                                 int endPosition) throws RemoteStorageException;
 
     /**
      * Returns the index for the respective log segment of {@link RemoteLogSegmentMetadata}.
+     * <p>
+     * If the index is not present (e.g. Transaction index may not exist because segments create prior to
+     * version 2.8.0 will not have transaction index associated with them.),
+     * throws {@link RemoteResourceNotFoundException}
      *
      * @param remoteLogSegmentMetadata metadata about the remote log segment.
      * @param indexType                type of the index to be fetched for the segment.
      * @return input stream of the requested index.
      * @throws RemoteStorageException          if there are any errors while fetching the index.
-     * @throws RemoteResourceNotFoundException when there are no resources associated with the given remoteLogSegmentMetadata.
+     * @throws RemoteResourceNotFoundException the requested index is not found in the remote storage.
+     * The caller of this function are encouraged to re-create the indexes from the segment
+     * as the suggested way of handling this error.
```

Comment:
This is confused because transactionIndex should be expected to be non-existed. Maybe we can put it this way:
`The caller of this function are encouraged to re-create the indexes from the segment as the suggested way of handling this error if the index is expected to be existed.`

WDYT?